### ISSUE TITLE
Resizing a window glitches if its content increases MinSize during the process

### ISSUE THREAD

**Issue by nullst** [fyne-io/fyne#4031]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Imagine that you are resizing a fyne application window by dragging the corner with the mouse. If an update to some widget in the window causes the window's `Content().MinSize()` to increase during the process of resizing, the window size will (momentarily) increase by the same amount, even if the window is easily large enough to accommodate the increased minimum size.

This looks especially strange if you stop moving the mouse but keep the mouse button down. Then each update to the minimum size will increase window's size, repeatedly.

### How to reproduce

1. Run the example code.
2. Start resizing the window by dragging the lower right corner.
3. After moving the cursor a little bit, stop moving the mouse, but keep the mouse button down.
4. Observe that the window keeps increasing its size every two seconds.
5. Note that if you start moving the mouse again, the size of the window is reset back to what you expect.

### Screenshots

 Here's how it looks with a simple application whose content is a single label which changes its length every second:

[resizing-jump.webm](https://github.com/fyne-io/fyne/assets/7746216/8686d9e3-6443-48b8-8271-35bc9707cde6)

### Example code

```go
package main

import (
	"time"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	application := app.NewWithID("resizing-bug")
	window := application.NewWindow("Resizing bug")

	label := widget.NewLabel("Short")

	window.Resize(fyne.NewSize(float32(200), float32(100)))
	window.SetContent(label)

	ticker := time.NewTicker(time.Second)
	n := 0
	go func() {
		for {
			<-ticker.C
			if n%2 == 0 {
				label.SetText("Short")
			} else {
				label.SetText("Very very long")
			}
			n++
		}
	}()

	window.ShowAndRun()
}
```

### Fyne version

2.3.5; also works on develop (bd682d1b93e2b25f90b998a3498cf46da6a246eb)

### Go compiler version

1.19.2

### Operating system and version

Linux Pop!_OS 21.10

**Comment by Jacalz** [comment]
I’m not necessarily saying that the issue is wrong but a MinSize that increases consistently seems a bit like an anti-pattern. You would probably be better off with adding some sort of wrapping to your label.

**Comment by Jacalz** [comment]
Where are you running into this outside of making labels longer?

If a layout (something custom etc.) increases the MinSize without changing content then I’d say that layout is probably using MinSize incorrectly. A minimum size (again, without changing content) that increases is not really a minimum anymore.

**Comment by andydotxyz** [comment]
I think this may be another instance of the Gnome resize issues we have seen during dragging between monitors...

I think this is #2704 - Gnome / Cinnamon exhibit strange resize issues that are not what we request of the window.
Other window managers and OS don't have this problem...

**Comment by nullst** [comment]
> I’m not necessarily saying that the issue is wrong but a MinSize that increases consistently seems a bit like an anti-pattern. You would probably be better off with adding some sort of wrapping to your label.

Yes, I agree, this is kind of a toy bug impacting mostly toy programs.

Still a bug, though: my expectation would be that if Content().MinSize() increases, but the window is large enough to fit the increased content, the window size should not change. Normally this works fine, but apparently if the MinSize() increase happens during the process of resizing the window, something strange happens. (At least on Linux, I don't have access to other platforms).

> If a layout (something custom etc.) increases the MinSize without changing content then I’d say that layout is probably using MinSize incorrectly. 

I suppose if we want to imagine a situation where this bug would surface for an actual application, then it would certainly be an application that updates its content automatically from time to time, based on network input or just the passage of time. Then the change in MinSize() could occur naturally, if the window's layout permits it. Shouldn't happen for nontrivial layouts, though.

**Comment by nullst** [comment]
I'm trying to experiment more with this issue, but it just keeps getting more and more mysterious. I wanted to find which part of the code is responsible for producing the wrong window size. I tried inserting print statements to various parts of fyne window handling code, so far no insights. I also tried debugging, but it's hard to debug event loops: I know that a `XConfigureEvent` with incorrect window sizes just arrives to the glfw X11 event handling code, but why and who sent it -- that I do not know, and I don't know how to figure out.

The only thing I found is that if I use the provided example code, grab window's lower right corner and hold the mouse button without moving the cursor, the window dimensions are always changed by exactly the same amount: the width is **always** increased by 20 and the height by 55. 

These two numbers **do not depend** on the initial window size, on the current window size, on the window position, on the length of window title, on the text of the labels, on the timings between minimum size changes, on the content of the window (e.g., putting an extra widget below the label doesn't change anything).

So, maybe someone can tell me. Is 20 x 55 a size of... something default in fyne? This is so confusing to me. It's not proportional to the size of the window. It's too vertical to be related to window decorations. Maybe scrollbars?... What can this be???

**Comment by andydotxyz** [comment]
> The only thing that confuses me is why the bug doesn't manifest on other systems. Maybe it does?

I'm pretty sure this is Linux specific, and may even be Gnome specific. It seems to relate to some X resize calls being respected and others ignored, somehow amplifying a grow loop.

> So, maybe someone can tell me. Is 20 x 55 a size of... something default in fyne?

I cannot think this would be - though if those are pixel sizes we would need to consider your scale value as well to know what number in Fyne equates to those pixel sizes.

**Comment by nullst** [comment]
Yeah, this issue seems GNOME-specific. I found what I assume is a different aspect of the same problem on the [mutter](https://gitlab.gnome.org/GNOME/mutter) bug tracker (mutter is a library gnome-shell uses for window management). The bug report is [here](https://gitlab.gnome.org/GNOME/mutter/-/issues/2890), and by a miraculous coincidence it was reported on the same day as this bug here. It demonstrates a similar problem using Motif GUI library, so probably nothing can be done in the Fyne code to work around this.

(As a final remark, I found a simpler way to demonstrate the issue I reported: just make a window that every second resizes itself to be 300x100. The window contents can be anything. Normally this works OK. Put the mouse cursor at the right lower corner of the window and press the mouse button down to start resizing, but don't move the cursor. Then, until you release the mouse button, the window will become larger and larger every second, just like in the video at the top post. This has nothing to do with MinSize(), just the fact that fyne resizes the window.)
